docs/containers-registries.conf.5.md

```diff
@@ -109,6 +109,31 @@ internet without having to change `Dockerfile`s, or to add redundancy).
 *Note*: Redirection and mirrors are currently processed only when reading images, not when pushing
 to a registry; that may change in the future.
 
+#### Mirroring and rewriting the docker.io registry
+
+The default Docker registry `docker.io` is handled in a special way, because
+every push and pull operation gets internally suffixed with `/library`, but only
+if no other specific repository is defined. This means that a pull of
+`docker.io/alpine` will be internally translated to `docker.io/library/alpine`.
+A pull of `docker.io/user/alpine` will not be rewritten because this is already
+the correct remote path.
+
+This behavior has to be considered when generally rewriting registries under
+the unqualified `docker.io` domain, because containers/image has to match
+Docker’s behavior in that case.
+
+For example, if a registry mirror is setup under `localhost/docker.io`,
+then the appropriate rewrite semantics would look as follows:
```

Comment:
This is not _always_ the right thing to do, e.g. when mirroring the users’ own Docker Hub images.

It’s great to have this example; this should be a bit clearer about when it is appropriate.